### PR TITLE
Add globalization

### PR DESCRIPTION
**Pull request by 0candy**
https://github.com/strongloop/loopback/issues/2422

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,6 @@ coverage
 *.xml
 .loopbackrc
 .idea
+!intl/
+intl/*
+!intl/en/
```
Comment (Amir-61) [inline .gitignore]: I would use two lines. Please see : https://github.com/strongloop-internal/loopback-knowledge-base/pull/51/files

File: index.js [hunk 2]
```diff
@@ -3,4 +3,7 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
+var SG = require('strong-globalize');
+SG.SetRootDir(__dirname);
+
```
Comment (Amir-61) [inline index.js]: One line would be neater. thought? Please see: https://github.com/strongloop-internal/loopback-knowledge-base/pull/51/files

File: lib/discovery.js [hunk 3]
```diff
@@ -3,6 +3,9 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
+var SG = require('strong-globalize');
+var g = SG();
```
Comment (Amir-61) [inline lib/discovery.js]: ditto

File: lib/discovery.js [hunk 4]
```diff
@@ -167,15 +170,15 @@ function mixinDiscovery(MySQL, mysql) {
    */
   function getArgs(table, options, cb) {
     if ('string' !== typeof table || !table) {
-      throw new Error('table is a required string argument: ' + table);
+      throw new Error(g.f('table is a required string argument: %s', table));
```
Comment (Amir-61) [inline lib/discovery.js]: should't be `{{table}}`?

File: lib/discovery.js [hunk 5]
```diff
@@ -167,15 +170,15 @@ function mixinDiscovery(MySQL, mysql) {
...
     }
     options = options || {};
     if (!cb && 'function' === typeof options) {
       cb = options;
       options = {};
     }
     if (typeof options !== 'object') {
-      throw new Error('options must be an object: ' + options);
+      throw new Error(g.f('options must be an object: %s', options));
```
Comment (Amir-61) [inline lib/discovery.js]: I used `{{options}}` and `{{object}}` in my PRs. I leave the decision up to you

File: lib/enumFactory.js [hunk 6]
```diff
@@ -3,6 +3,9 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
+var SG = require('strong-globalize');
+var g = SG();
```
Comment (Amir-61) [inline lib/enumFactory.js]: ditto

File: lib/mysql.js [hunk 7]
```diff
@@ -3,6 +3,9 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
+var SG = require('strong-globalize');
+var g = SG();
```
Comment (Amir-61) [inline lib/mysql.js]: ditto

File: lib/mysql.js [hunk 8]
```diff
@@ -164,7 +167,7 @@ MySQL.prototype.executeSQL = function(sql, params, options, callback) {
   var debugEnabled = debug.enabled;
   var db = this.settings.database;
   if (typeof callback !== 'function') {
-    throw new Error('callback should be a function');
+    throw new Error(g.f('callback should be a function'));
```
Comment (Amir-61) [inline lib/mysql.js]: `{{callback}}`

File: lib/discovery.js (hunk 5 above)
Comment (jannyHou) [inline lib/discovery.js]: {{options}}?

File: lib/mysql.js [hunk 9]
```diff
@@ -517,13 +520,13 @@ MySQL.prototype.buildExpression = function(columnName, operator, operatorValue,
     propertyDefinition) {
   if (operator === 'regexp') {
     if (operatorValue.ignoreCase)
-      console.warn('MySQL regex syntax does not respect the `i` flag');
+      g.warn('{{MySQL}} regex syntax does not respect the {{`i`}} flag');
```
Comment (Amir-61) [inline lib/mysql.js]: I dont think if regex needs to be translated either; I would say '{{MySQL regex}}'

**Comment by jannyHou** [comment]
only one comment about wrapping `{{options}}`. LGTM 👍

File: lib/mysql.js [hunk 10]
```diff
@@ -517,13 +520,13 @@ MySQL.prototype.buildExpression = function(columnName, operator, operatorValue,
...
 
     if (operatorValue.global)
-      console.warn('MySQL regex syntax does not respect the `g` flag');
+      g.warn('{{MySQL}} regex syntax does not respect the {{`g`}} flag');
 
     if (operatorValue.multiline)
-      console.warn('MySQL regex syntax does not respect the `m` flag');
+      g.warn('{{MySQL}} regex syntax does not respect the {{`m`}} flag');
```
Comment (Amir-61) [inline lib/mysql.js]: ditto

**Comment by Amir-61** [comment]
@0candy reviewed; added a bunch of nitpicks...

File: lib/migration.js [hunk 11]
```diff
@@ -3,6 +3,8 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
+var g = require('strong-globalize')();
+
 var async = require('async');
```
Comment (Amir-61) [inline lib/migration.js]: Nitpick: What is this extra line between variables?

File: .gitignore (hunk 1 above)
Comment (Amir-61) [inline .gitignore]: Nitpick: We usually leave one blank line at the end of file.

**Comment by Amir-61** [comment]
Two minor comments which feel free to ignore. LGTM

File: intl/en/messages.json [hunk 12]
```diff
@@ -0,0 +1,10 @@
+{
+  "1092fa6c86f124928e0f4d9d4711ebc9": "options must be an object: {0}",
+  "66a6a44b45094ef45fdcb0201a26fc6d": "table is a required string argument: {0}",
+  "b7c60421de706ca1e050f2a86953745e": "No arguments - could not create {{Enum}}.",
+  "80a32e80cbed65eba2103201a7c94710": "Model not found: {0}",
+  "610db3e4d1e877d985000ed09fb01fb9": "callback should be a function",
+  "a354e762822e356af4bfa3299fd29b0a": "{{MySQL}} regex syntax does not respect the {{`g`}} flag",
+  "cecffcfbf707d97548dcd5859c80b171": "{{MySQL}} regex syntax does not respect the {{`i`}} flag",
+  "d230064cbc14fe9bfcf2fb2072c16fb2": "{{MySQL}} regex syntax does not respect the {{`m`}} flag"
```
Comment (Amir-61) [inline intl/en/messages.json]: you forgot to to do `slt-globalize -e` and `slt-globalize -l` to update this file?

**Comment by rmg** [comment]
@slnode test please

**Pull request merged**